mock_service.go

```diff
@@ -31,33 +32,58 @@ type MockEndpoint struct {
 	// RequestBody     string            `json:"requestBody"`
 }
 
-func New(createMockEndpoint string) *MockService {
+type MockEndpointSeries []MockEndpoint
+
+func New(mockCreatorEndpoint string) *MockService {
 	return &MockService{
-		createMockEndpoint: createMockEndpoint,
-		mockEndpoints:      map[string]map[string]MockEndpoint{},
+		mockCreatorEndpoint:  mockCreatorEndpoint,
+		mockedEndpoints:      map[string]map[string]MockEndpoint{},
 	}
 }
 
 func (m *MockService) ServeHTTP(w http.ResponseWriter, req *http.Request) {
-	if req.Method == http.MethodPost && req.URL.Path == m.createMockEndpoint {
+	if req.Method == http.MethodPost && req.URL.Path == m.mockCreatorEndpoint {
 		m.serveRegistrationHTTP(w, req)
 		return
 	}
 
 	m.serveMockHTTP(w, req)
 }
 
+func (m *MockService) PreloadEndpointsFromConf(confFilePath string) (error, bool){
+    if  len(m.mockCreatorEndpoint) < 1 {
+        return  ErrCreatorNotExist, false
+    }
+    fi, err := os.Open(confFilePath)
```

Comment:
Instead of the mock service opening the file, perhaps the constructor should take an argument of the pointer to some struct that defines the configuration file that is formatted in YAML, JSON, or so on. If its defined (non-nil) we can preload it, otherwise it'll create an empty `mockEndpoints` in the `*MockService*` returned.

That was the `io` could be unopinionated eg. the configuration could be read from somewhere else other than a file and passed in by the caller instead of the `mock_service` library defining where the configuration is coming from.